packages/bpk-component-banner-alert/src/withBannerAlertState.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import { wrapDisplayName } from 'bpk-react-utils';
+
+const withBannerAlertState = (WrappedComponent) => {
+  class component extends Component {
+    constructor(props) {
+      super(props);
+
+      this.state = {
+        expanded: props.expanded,
+        dismissed: false,
+      };
+
+      this.onExpandToggle = this.onExpandToggle.bind(this);
+      this.onDismiss = this.onDismiss.bind(this);
+    }
+
+    componentWillMount() {
+      const { hideAfter } = this.props;
+
+      if (hideAfter && hideAfter > 0) {
+        this.hideIntervalId = setTimeout(() => {
+          this.onDismiss();
+        }, hideAfter * 1000);
+      }
+    }
+
+    componentWillUnmount() {
+      if (this.hideIntervalId) {
+        clearTimeout(this.hideIntervalId);
+      }
+    }
+
+    onExpandToggle() {
+      const expanded = !this.state.expanded;
+      this.setState({ expanded });
+
+      if (this.props.onExpandToggle) {
```

Comment:
I've added the option to bind callbacks when expanded/collapsed/dismissed, in case the user of the component needs to be notified to perform some other action on their side.